tests/Struct/FindologicProductTest.php

```diff
@@ -379,6 +379,72 @@ public function testProductRatings(array $ratings, float $expectedRating): void
         $this->assertEquals($expectedRating, current($ratingAttribute->getValues()));
     }
 
+    public function attributeProvider(): array
+    {
+        return [
+            'filter with some special characters' => [
+                'attributeName' => 'Special Characters /#+*()()=§(=\'\'!!"$.|',
+                'expectedName' => 'SpecialCharacters'
+            ],
+            'filter with brackets' => [
+                'attributeName' => 'Farbwiedergabe (Ra/CRI)',
+                'expectedName' => 'FarbwiedergabeRaCRI'
+            ],
+            'filter with special UTF-8 characters' => [
+                'attributeName' => 'Ausschnitt D ø (mm)',
+                'expectedName' => 'AusschnittDmm'
+            ],
+            'filter dots and dashes' => [
+                'attributeName' => 'free_shipping.. Really Cool--__',
+                'expectedName' => 'free_shippingReallyCool--__'
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider attributeProvider
+     */
+    public function testAttributesAreProperlyEscaped(string $attributeName, string $expectedName): void
+    {
+        $productEntity = $this->createTestProduct([
+            'properties' => [
+                [
+                    'id' => Uuid::randomHex(),
+                    'name' => 'some value',
+                    'group' => [
+                        'id' => Uuid::randomHex(),
+                        'name' => $attributeName
+                    ],
+                ]
+            ]
+        ]);
+
+        $criteria = new Criteria([$productEntity->getId()]);
+        $criteria = Utils::addProductAssociations($criteria);
+
+        $productEntity = $this->getContainer()->get('product.repository')->search($criteria, $this->defaultContext)
+            ->get($productEntity->getId());
+
+        $customerGroupEntities = $this->getContainer()
+            ->get('customer_group.repository')
+            ->search(new Criteria(), $this->defaultContext)
+            ->getElements();
+
+        $findologicProductFactory = new FindologicProductFactory();
+        $findologicProduct = $findologicProductFactory->buildInstance(
+            $productEntity,
+            $this->router,
+            $this->getContainer(),
+            $this->defaultContext,
+            $this->shopkey,
+            $customerGroupEntities,
+            new XMLItem('123')
+        );
+
+        $this->assertCount(5, $findologicProduct->getAttributes());
```

Comment:
Why 5?

Reply:
Because there are several other attributes like `cat_url` or `vendor`. All in all, this are 5 :).

Reply:
Where do I see all the other attributes? In the test one property gets set.

Reply:
They are automatically created since the product needs to have some data to even be created in the first place. The method `ProductHelper::createTestProduct` takes care of that.

https://github.com/findologic/plugin-shopware-6/blob/96099d5de1152caba36f7bd7f1e979c5e50dff8e/tests/Struct/FindologicProductTest.php#L413

Reply:
Ok I understand but the numer of attributes is irrelevant for the test. Can you please change the assertion so we check if `$attributeName` is in the properties?